tools/scripts/lock-api-state.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash -ex
+
+# Buf image binary
+BUF_IMAGE=buf.image.bin
+PROJECT_ROOT=$(pwd)
+
+## lock in current protobuf state
+cd proto
+make gen-buf-image
+git add $BUF_IMAGE
+git commit -m "chore: lock api state for backward compatibility check" || echo "buf image is already up to date"
```

Comment:
I think this `||` could catch too many different errors.  How about you run another `git status --porcelain` test and exit early if there is no change.

Then if there is a change, you require that `git commit` passes.